### PR TITLE
Completed express-crud exercise

### PR DESCRIPTION
**Pull request by techmatlock**
I'm not sure if I specified type in the right area but I type'd the req object in the async function.

### CODE REVIEW

File: express-crud/server.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  try {
+    const { gradeId } = req.params;
+    if (!Number.isInteger(+gradeId))
+      throw new ClientError(400, `gradeId ${gradeId} must be a number.`);
+    if (!gradeId) throw new ClientError(400, `gradeId ${gradeId} is required.`);
+
+    const { name, course, score } = req.body;
+    if (!name || !course || !score)
+      throw new ClientError(400, 'Invalid/missing name, course, or score.');
+
```
Comment (shawnkost) [inline express-crud/server.ts:70]: What if they gave us a score that's less than 0 or greater than a 100? We should check for that as well

File: express-crud/server.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        set "name" = $1,
+            "course" = $2,
+            "score" = $3
+        where "gradeId" = $4
+        returning *;
+    `;
+
+    const params = [name, course, score, gradeId];
+    const result = await db.query<Grade>(sql, params);
+    if (!result) res.sendStatus(500);
```
Comment (shawnkost) [inline express-crud/server.ts:82]: Not needed

File: express-crud/server.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+    const sql = `
+      delete from "grades"
+        where "gradeId" = $1
+        returning *;
+    `;
+
+    const params = [gradeId];
+    const result = await db.query<Grade>(sql, params);
+    if (!result) res.sendStatus(500);
```
Comment (shawnkost) [inline express-crud/server.ts:110]: Not needed

**Review by shawnkost — CHANGES_REQUESTED** [review]
Left some comments here about optimizations, but once you add a `post` endpoint this is good to go!

File: express-crud/server.ts (hunk 1 above)
Comment (techmatlock) [reply express-crud/server.ts:70]: Good catch.  Just added conditional.

File: express-crud/server.ts [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    select *
+      from "grades"
+      where "gradeId" = $1;
+  `;
+
+    const params = [gradeId];
+    const result = await db.query(sql, params);
+    if (!result) res.sendStatus(500);
+    const grade = result.rows[0];
+    if (!grade) res.status(404).send('File not found.');
```
[the remark this answers is not in the document]
Comment (techmatlock) [reply express-crud/server.ts:53]: Gtk.  Thanks.

**Review by shawnkost — APPROVED** [review]
👍🏻

**Pull request merged**